var/spack/repos/builtin/packages/perl-bioperl/package.py

```diff
@@ -32,87 +32,82 @@ class PerlBioperl(PerlPackage):
     and contribute your own if possible."""
 
     homepage = "https://metacpan.org/pod/BioPerl"
-    url = "https://cpan.metacpan.org/authors/id/C/CD/CDRAUG/BioPerl-1.7.6.tar.gz"
+    url = "https://cpan.metacpan.org/authors/id/C/CJ/CJFIELDS/BioPerl-1.7.8.tar.gz"
 
     license("Artistic-1.0")
 
+    version("1.7.8", sha256="c490a3be7715ea6e4305efd9710e5edab82dabc55fd786b6505b550a30d71738")
     version(
         "1.7.6",
         sha256="df2a3efc991b9b5d7cc9d038a1452c6dac910c9ad2a0e47e408dd692c111688d",
-        preferred=True,
+        url="https://cpan.metacpan.org/authors/id/C/CD/CDRAUG/BioPerl-1.7.6.tar.gz",
     )
+    version("1.7.2", sha256="17aa3aaab2f381bbcaffdc370002eaf28f2c341b538068d6586b2276a76464a1")
```

Comment:
While this is the same as 1.007002 on the Perl side, this does not work in Spack - it can not find this version.
I would recommend to just remove it.

You can see the issue when trying to validate checksums. When installing, the issue is usually not visible, since the sources are fetched through the checksum from the spack souce cache and there is source available for this checksum (because of 1.007002).
It seems, this is why this worked for me so far.

Reply:
Also, sorry for misleading you here with my suggested code.

Reply:
No problem. As I was looking through this I noticed what the intention was. In my refactoring, I overlooked the issue with the 1.7.2 version. Thanks for catching it!